modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/IgniteMdSelectivity.java

```diff
@@ -108,7 +110,7 @@ private static double computeOpsSelectivity(Map<RexNode, List<SqlKind>> operands
      * OR(<($t3, 110), >($t3, 150), AND(>=($t2, -($t1, 2)), <=($t2, +($t3, 2))), >($t4, $t2), <($t4, $t3)) selectivity computes separately
      * for each local ref with AND selectivity adjustment. <br>
      */
-    private static double computeOrSelectivity(RexCall call, @Nullable BitSet primaryKeys) {
+    private static double computeOrSelectivity(RexCall call, @Nullable BitSet primaryKeys, @Nullable Mapping columnMapping) {
```

Comment:
this fix is not related to issue at all, i just found it in progress of working, i just forget about mapping, additional tests are appended.
I can move it into different issue but probably all ok to stay it here ?

Reply:
I suggest to move unrelated changed to a separate PR, 
because the current ticket may be declined or postponed due to any reason,
and this should NOT affect the valid issue you found.

Reply:
moved into additional issue